### PR TITLE
Add access methods to Nes

### PR DESCRIPTION
**Pull request by johnor**
Makes it easier for a gui to access debug info

### CODE REVIEW

File: nes/CMakeLists.txt [hunk 1]
```diff
@@ -26,8 +26,9 @@ set_target_properties(${PROJECT_NAME}
 )
 
 target_link_libraries(${PROJECT_NAME}
-    PRIVATE
+    PUBLIC
```
Comment (robinlinden) [inline nes/CMakeLists.txt]: Maybe forward-declare the core components and leave this private? Then it's up to the consumer of the n_e_s::nes library if they want to pull in n_e_s::core to access e.g. the mmu.

Comment (johnor) [reply nes/CMakeLists.txt]: Good idea, but we can't do it right now. Not possible to forward declare nested classes (ICpu::Registers). We need to split those classes then.
There is a paper (P0289R0) to add support for this into the standard, but not sure about the status. It is not in c++20 at least.

Comment (johnor) [reply nes/CMakeLists.txt]: I can make Registers its own class if you want :)

Comment (robinlinden) [reply nes/CMakeLists.txt]: Rip. I'd be happy to move that class out of ICPu to keep Core and Nes separate. :D

Comment (johnor) [reply nes/CMakeLists.txt]: Done :D

**Pull request merged**